### DISCUSSION TITLE
Spotify steals my focus!

### DISCUSSION THREAD

**Discussion by smitropoulos** [i3/i3#5816] [Q&A]
Hello!

A couple days ago spotify-launcher (https://archlinux.org/packages/extra/x86_64/spotify-launcher/) had an update and I noticed that now spotify steals my focus every now and then!

I have 

```
# Not auto focus steal!
no_focus [class="Spotify"]
focus_on_window_activation urgent
```
on my config, and I have tried all the other modes for `focus_on_window_activation` with no luck. 

Has anyone else encountered this?

**Comment by orestisfl** [comment]
Do you mean that spotify steal focus while already being open or does it do that when starting up? `no_focus` is only applicable when the window is opening but Spotify is known to assign its class after mapping the window, so the `no_focus` rule will not work here.

A workaround is to throw Spotify on a dedicated workspace with a `for_window` rule, e.g. `for_window [class=Spotify] move to workspace number 10`

**Reply by smitropoulos** [reply]
Thank you for the reply, I apologize I was not transparent on the issue!

The thing is that below, I do move the window the exact number you said, number 10.

```
for_window [class="Spotify"] move to workspace $ws10, workspace --no-auto-back-and-forth $ws10
```

It keeps stealing focus when playing, I think when the song changes but I have also noticed it does it randomly, moving me to its workspace, which is quite annoying!

**Reply by orestisfl** [reply]
I imagine it is because of the `workspace --no-auto-back-and-forth $ws10` part. For some reason, the `for_window` rule is triggered many times and the workspace change part is re-triggered.

A hacky workaround would be:
```
for_window [class="Spotify"] move to workspace $ws10, exec "i3-msg -t get_marks | grep -v spotify && i3-msg mark spotify, workspace $ws10"
```
I haven't tested it but it should only focus the workspace when the `spotify` tag doesn't exist.

That said, that shouldn't happen. Can you upload a log file perhaps? https://i3wm.org/docs/debugging.html
